helix-core/src/main/java/org/apache/helix/controller/rebalancer/strategy/AutoRebalanceStrategy.java

```diff
@@ -560,7 +560,7 @@ private Map<Replica, Node> computeExistingPreferredPlacement(
           Replica replica = new Replica(partition, replicaId);
           if (_preferredAssignment.containsKey(replica)
               && !existingPreferredAssignment.containsKey(replica)
-              && _preferredAssignment.get(replica).id == node.id) {
+              && _preferredAssignment.get(replica).id == node.id && node.isAlive) {
```

Comment:
discussed offline. Need refine behavior consistent with current delay auto rebalanced. It means when an instance disabled, it should still show in preference list but in map field it showed as INITIAL state.